components/tidb_query_datatype/src/codec/mysql/time/mod.rs

```diff
@@ -2714,6 +2721,23 @@ mod tests {
         Ok(())
     }
 
+    #[test]
+    fn test_from_local_time() -> Result<()> {
+        let mut ctx = EvalContext::default();
+        for i in 2..10 {
+            let actual = Time::from_local_time(&mut ctx, TimeType::DateTime, i % MAX_FSP)?;
+            let c_datetime = actual.try_into_chrono_datetime(&mut ctx)?;
+
+            //2022-02-15 07:45:48.581699 NonFixed(UTC)   c_datetime
+            //2022-02-15 07:45:48.581933075 UTC          Utc::now()
+            //We compare length 22 of then; Exceed it may be not equal ( in milli seconds ).
+            let now0 = &c_datetime.to_string()[0..22];
+            let now1 = &Utc::now().to_string()[0..22];
```

Comment:
This can be freaky as time may change. Tests running at second boundary will fail.

Reply:
One of the failure cases: https://ci.pingcap.net/blue/organizations/jenkins/tikv_ghpr_test/detail/tikv_ghpr_test/17038/pipeline/116

Reply:
+1 for revert this

Reply:
I may add a range to fix it.

Reply:
Note second changes can even lead to year changes.